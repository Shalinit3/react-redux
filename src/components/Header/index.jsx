/**
 * Master Header component will be used throught out the pages.
 */

import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'

import logo from '../../images/logo.svg'

class Header extends Component {
  render () {
    let {visibility: { topVisible = true } = {}} = this.props

    return (
      <Menu className={topVisible ? 'visible' : ''} borderless size='massive' fixed='top'>
        <Menu.Item as='a' header>
          <img src={logo} alt='Robokind' />
        </Menu.Item>

        <Menu.Item
          name='sign-in'
          position='right'
        >
            Sign-in
        </Menu.Item>
      </Menu>
    )
  }
}

export default Header
