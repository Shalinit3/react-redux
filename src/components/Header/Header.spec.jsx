import React from 'react'
import { shallow } from 'enzyme'

import Header from './index'

describe('Header', () => {
  let sut

  beforeEach(() => {
    sut = shallow(<Header />)
  })

  it(`should match header snapshot`, () => {
    expect(sut.find('Menu').get(0)).toMatchSnapshot()
  })

  describe('should have prop className as visible', () => {
    let visibility = {
      topVisible: true
    }

    beforeEach(() => {
      sut.setProps({visibility})
    })

    it(`should not have visibility class`, () => {
      expect(sut.find('Menu').prop('className')).toBe('visible')
    })
  })

  describe('should have prop className as blank', () => {
    let visibility = {
      topVisible: false
    }

    beforeEach(() => {
      sut.setProps({visibility})
    })

    it(`should not have visible class `, () => {
      expect(sut.find('Menu').prop('className')).toBe('')
    })
  })
})
