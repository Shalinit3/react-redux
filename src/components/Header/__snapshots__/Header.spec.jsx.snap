// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match header snapshot 1`] = `
<Menu
  borderless={true}
  className="visible"
  fixed="top"
  size="massive"
>
  <MenuItem
    as="a"
    header={true}
  >
    <img
      alt="Robokind"
      src="logo.svg"
    />
  </MenuItem>
  <MenuItem
    name="sign-in"
    position="right"
  >
    Sign-in
  </MenuItem>
</Menu>
`;
